scripts/tso-enum.nse

```diff
@@ -165,35 +173,89 @@ local function tso_test( host, port, commands )
     stdnse.debug("Could not initiate TN3270: %s", err )
     return tso, "Could not Initiate TN3270"
   end
-  stdnse.debug("Getting to TSO")
   local run = stdnse.strsplit(";%s*", commands)
   for i = 1, #run do
-    stdnse.debug(1,"Issuing Command (#%s of %s): %s", i, #run ,run[i])
+    stdnse.debug(2,"Issuing Command (#%s of %s): %s", i, #run ,run[i])
     tn:send_cursor(run[i])
     tn:get_all_data()
   end
   tn:get_screen_debug(2)
 
+  if tn:find("***") then
+    secprod = "TopSecret/ACF2"
+  end
+
   if tn:find("ENTER USERID") or tn:find("TSO/E LOGON")  then
     tso = true
-    -- Patch OA44855 removed the ability to enumerator users
-    -- we check for that here
+    -- Patch OA44855 removed the ability to enumerate users
     tn:send_cursor("notreal")
     tn:get_all_data()
     if tn:find("IKJ56476I ENTER PASSWORD") then
-      return false, "Could not enumerate. PASSWORDPREPROMPT is set to ON."
+      return false, secprod, "Enumeration is not possible. PASSWORDPREPROMPT is set to ON."
     end
   end
-
-  if tn:find("***") then
-    secprod = "TopSecret/ACF2"
-  end
-
   tn:send_pf(3)
   tn:disconnect()
   return tso, secprod, "Could not get to TSO. Try --script-args=tso-enum.commands='logon applid(tso)'. Aborting."
 end
 
+--- Tests the target to see if we can speed up brute forcing
+-- VTAM/USSTable will sometimes allow you to put the userid
+-- in the command area either through data() or just adding
+-- the userid. This function will test for both
+-- 
+-- @param host host NSE object
+-- @param port port NSE object
+-- @param commands script-args of commands to use to get to TSO
+-- @return status true on success, false on failure
+local function tso_skip( host, port, commands )
```

Comment:
Would it make sense to put this function into `tn3270.lua` since you call it in several places? It could even cache results in the [`host.registry`](https://nmap.org/book/nse-api.html) to avoid re-testing if you run multiple tso scripts at once.